src/multiplayer.h

```diff
@@ -49,6 +49,14 @@ template <typename T> struct multiplayerReplicationFunctions
 {
     static bool isChanged(void* data, void* prev_data_ptr)
     {
+        if ((data == nullptr) != (prev_data_ptr == nullptr)) {
```

Comment:
This whole check looks odd, both pointers should never be nullptr, that can be an assert.

Reply:
Unfortunately I found that `prev_data_ptr` can be `nullptr` (before this commit it was set to `-1`). Honestly I did not check if `data` can be a null pointer, that could be asserted. If `prev_data_ptr` is `nullptr` the function should just return `true` or it should initialize `prev_data_ptr` with a copy of `data` and return `true`?

Reply:
I think you are mixing up prev_data and prev_data_ptr. As prev_data_ptr cannot be null, the code would crash if it was, and it is always set to &prev_data.

Reply:
My fault, you are right: I missed that `prev_data` is not casted to a pointer, but the pointer to `prev_data` is passed.

Unfortunately the UB is still there (and I need to change my PR), it is just different: `prev_data` cannot be casted to `T` in some cases because the bit pattern is invalid -- triggering UB.

I will try to come up with something meaningful to handle the situation, if you have any suggestion just write me. :wink: